processing/src/main/java/org/apache/druid/segment/ColumnarValueUnnestCursor.java

```diff
@@ -0,0 +1,270 @@
+      {
+        if (!outputName.equals(columnName)) {
+          return baseColumSelectorFactory.makeColumnValueSelector(columnName);
+        }
+        return new ColumnValueSelector()
+        {
+          @Override
+          public double getDouble()
+          {
+            Object value = getObject();
+            if (value == null) {
+              return 0;
+            }
+            return Double.valueOf((String) value);
+          }
+
+          @Override
+          public float getFloat()
+          {
+            Object value = getObject();
+            if (value == null) {
+              return 0;
+            }
+            return Float.valueOf((String) value);
+          }
+
+          @Override
+          public long getLong()
+          {
+            Object value = getObject();
+            if (value == null) {
+              return 0;
+            }
+            return Long.valueOf((String) value);
+          }
+
+          @Override
+          public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+          {
+            baseColumSelectorFactory.makeColumnValueSelector(columnName).inspectRuntimeShape(inspector);
+          }
+
+          @Override
+          public boolean isNull()
+          {
+            return getObject() == null;
+          }
+
+          @Nullable
+          @Override
+          public Object getObject()
+          {
+            if (!unnestListForCurrentRow.isEmpty()) {
+              if (allowSet == null || allowSet.isEmpty()) {
+                return unnestListForCurrentRow.get(index);
+              } else if (allowSet.contains((String) unnestListForCurrentRow.get(index))) {
+                return unnestListForCurrentRow.get(index);
+              }
+            }
+            return null;
+          }
+
+          @Override
+          public Class classOfObject()
+          {
+            return Object.class;
+          }
+        };
+      }
+
+      @Nullable
+      @Override
+      public ColumnCapabilities getColumnCapabilities(String column)
+      {
+        if (!outputName.equals(columnName)) {
+          baseColumSelectorFactory.getColumnCapabilities(column);
+        }
+        return baseColumSelectorFactory.getColumnCapabilities(columnName);
+      }
+    };
+  }
+
+  @Override
+  public DateTime getTime()
+  {
+    return baseCursor.getTime();
+  }
+
+  @Override
+  public void advance()
+  {
+    advanceUninterruptibly();
+    BaseQuery.checkInterrupted();
+  }
+
+  @Override
+  public void advanceUninterruptibly()
+  {
+    do {
+      advanceAndUpdate();
+    } while (matchAndProceed());
+  }
+
+  @Override
+  public boolean isDone()
+  {
+    if (needInitialization && baseCursor.isDone() == false) {
+      initialize();
+    }
+    return baseCursor.isDone();
+  }
+
+  @Override
+  public boolean isDoneOrInterrupted()
+  {
+    if (needInitialization && baseCursor.isDoneOrInterrupted() == false) {
```

Comment:
`==` on a boolean thing is a bit weird?  Just `!baseCursor.isDoneOrInterrupted()`?